test/services/organizations/create_service_test.rb

```diff
@@ -21,7 +20,21 @@ class Organizations::CreateServiceTest < ActiveSupport::TestCase
       }
     }
 
+    Organizations::CreateService.any_instance.stubs(:send_email).returns(true)
+
     Organizations::CreateService.new.signal(args)
```

Comment:
In the context of this test, we are making assertions on `Organization.last` but from reading the test alone we have no indication if the organization actually gets created. There could be an Organization in the fixture data for example that is not being created by this service when we call it. So, how about we wrap this in a block that will validate the question 'is an organization being created?'
```
assert_difference "Organization.count", 1 do
  create the organization
  rest of the assertions (but we can get rid of assert_not_nil assertions, now)
end

does this make sense?


```

Reply:
I believe so. Correct me if I'm wrong but the worry is that by relying on Organization.last there is a potential that we are not relying an a newly created test organization but an actual one which could potentially eschew the results and expectations of our logic in the test, right? So you want me to implement some logic to check to see if a test organization is actually being generated and then proceed with the rest of the test which allows us to remove the assert_not_nil assertions as well since we should know we'd be acting on an organization that has just been created. I'll get to work on it.

Reply:
Yes that's correct!

Reply:
Cool! I just pushed up the changes and have it all wrapped in the `assert_difference`. I did keep the `assert_not_nil`s for the location value and custom_page in case there is some other flawed logic or other reason those values are not made in the organization. But I see that my checks did not succeed and I think it's due to some spacing mistakes. I'll quickly rectify that!